sdk/src/main/java/io/opentelemetry/sdk/tags/TagMapSdk.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.tags;
+
+import static io.opentelemetry.internal.Utils.checkNotNull;
+
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.tags.Tag;
+import io.opentelemetry.tags.TagKey;
+import io.opentelemetry.tags.TagMap;
+import io.opentelemetry.tags.TagMetadata;
+import io.opentelemetry.tags.TagValue;
+import io.opentelemetry.tags.unsafe.ContextUtils;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public class TagMapSdk implements TagMap {
+
+  // The types of the TagKey and Tag must match for each entry.
+  private final Map<TagKey, Tag> tags;
+  @Nullable private final TagMap parent;
+
+  /**
+   * Creates a new {@link TagMapSdk} with the given tags.
+   *
+   * @param tags the initial tags for this {@code TagMapSdk}.
+   * @param parent providing a default set of tags
+   */
+  private TagMapSdk(Map<? extends TagKey, ? extends Tag> tags, TagMap parent) {
+    this.tags = Collections.unmodifiableMap(new HashMap<>(checkNotNull(tags, "tags")));
+    this.parent = parent;
+  }
+
+  @Override
+  public Iterator<Tag> getIterator() {
+    Map<TagKey, Tag> combined = new HashMap<>();
+    if (parent != null) {
+      for (Iterator<Tag> it = parent.getIterator(); it.hasNext(); ) {
+        Tag tag = it.next();
+        combined.put(tag.getKey(), tag);
```

Comment:
The current tags are newer than the old once. So you need to put only if not present.